RNTester/js/TimerExample.js

```diff
@@ -117,28 +116,31 @@ class RequestIdleCallbackTester extends React.Component<{}, $FlowFixMeState> {
   };
 }
 
-var TimerTester = createReactClass({
-  displayName: 'TimerTester',
+type TimerTesterProps = $ReadOnly<{|
+  dt?: number,
+  type: string,
+|}>;
 
-  _ii: 0,
-  _iters: 0,
-  _start: 0,
-  _timerId: (null: ?TimeoutID),
-  _rafId: (null: ?AnimationFrameID),
-  _intervalId: (null: ?IntervalID),
-  _immediateId: (null: ?Object),
-  _timerFn: (null: ?() => any),
+class TimerTester extends React.Component<TimerTesterProps> {
+  _ii = 0;
+  _iters = 0;
+  _start = 0;
+  _timerId = (null: ?TimeoutID);
+  _rafId = (null: ?AnimationFrameID);
+  _intervalId = (null: ?IntervalID);
+  _immediateId = (null: ?Object);
+  _timerFn = (null: ?() => any);
 
-  render: function() {
+  render() {
     var args = 'fn' + (this.props.dt !== undefined ? ', ' + this.props.dt : '');
     return (
       <RNTesterButton onPress={this._run}>
         Measure: {this.props.type}({args}) - {this._ii || 0}
       </RNTesterButton>
     );
-  },
+  }
 
-  componentWillUnmount() {
+  componentWillUnmount = () => {
```

Comment:
There's no need to bind component life-cycle hooks. Since React is responsible for calling these hooks, it can make sure that the value of `this` is correct inside them.

Reply:
You are completely right! Here is the change [18ff98b](https://github.com/facebook/react-native/pull/21623/commits/18ff98bfcd27d74afe2cc6b5bff93ed2af6d004e)